### PR TITLE
PDI-2073: Add adapter name to EA response

### PR DESCRIPTION
**Pull request by finleydecker**
- added adapterName field to the response.meta object in cache/response.ts
- created an AdapterResponseMeta type
- added a test to validate the response.meta object

### CODE REVIEW

File: test/metrics/metrics.test.ts [hunk 1]
```diff
@@ -297,3 +297,21 @@ test('invalid metric name throws error', async (t) => {
...
+  axiosMock
+    .onGet(`${URL}${endpoint}`, {
+      base: from,
+      quote: to,
+    })
+    .reply(200, {
+      price,
+    })
+
+  const response = await (await t.context.testAdapter.request({ from, to })).json()
```
[the remark this answers is not in the document]
Comment (finleydecker) [reply test/metrics/metrics.test.ts:311]: yep it does!

https://developer.mozilla.org/en-US/docs/Web/API/Response/json

[the remark this answers is not in the document]
Comment (finleydecker) [reply test/metrics/metrics.test.ts:311]: Screenshot 2023-04-20 at 5 42 52 AM

[the remark this answers is not in the document]
Comment (amit-momin) [reply test/metrics/metrics.test.ts:311]: I think you'd still need to wrap the request with the await but the `LightMyRequest` json() doesn't return a promise. So something like 
```suggestion
  const response = (await t.context.testAdapter.request({ from, to })).json()
```

[the remark this answers is not in the document]
Comment (finleydecker) [reply test/metrics/metrics.test.ts:311]: ahh i see. ok will fix that quickly! thanks!

**Review by amit-momin — APPROVED** [review]

**Pull request merged**